doc/faq.rst

```diff
@@ -34,12 +34,15 @@ An ECC key is roughly 500 bytes formatted in Base64 and RSA 2048 key is
 we opted for readability.
 
 
-Why do you drop all headers if there is more than one?
-------------------------------------------------------
-
-Because of multi-agent usage we may have to handle an inconsistent stream of
-headers already. Making this an inconsistent stream of multiple keys with
-priorities sounds like a lot of pain.
+Why do you ask to drop all headers if there is more than one?
+-------------------------------------------------------------
+
+Because of multi-agent usage we may have to handle an inconsistent
+stream of headers already. Making this an inconsistent stream of
+multiple headers per mail sounds like a lot of pain. We could come up
```

Comment:
@hpk42 , the problem is that the overall behaviour of the system becomes unpredictable. That is never a desirable quality. To avoid it being unpredictable, we would need to clearly specify the rules for choosing a header, which adds complexity. Also not a desirable thing.

Making it clear that this is a spec violation and ignoring it completely is simple and in the exact same spirit as our rules about parsing the headers themselves: if mandatory attributes are seen that we don't recognize, we ignore the header because we cannot be confident that we're doing the right thing. The same logic applies here: if we see multiple headers, we cannot be confident that choosing any particular one is the right thing - so we error out and do nothing.

It's clear. Once we have more experience from the field we'll know *how* this fails and at that point we'll be in a position to draft more complicated rules. But at this early stage, we just don't know enough and should strive to keep things simple.

In my opinion, obviously... ;-)